src/components/steps/save/SaveStep.tsx

```diff
@@ -57,18 +58,21 @@ const SaveStep: React.FC<StepProps> = ({ project, onCancel, action }) => {
     );
 
     const checkExistingDataAndSave = React.useCallback(async () => {
+        loading.show(true, i18n.t("Validating Project"));
         const validation = await project.validate();
         const error = _(validation).values().flatten().join("\n");
         if (error) {
             snackbar.error(error);
             return;
         }
 
+        loading.updateMessage(i18n.t("Checking Existing Data"));
         const existingDataCheck = await project.checkExistingDataForDataElementsToBeRemoved();
 
+        loading.updateMessage(i18n.t("Saving Project"));
         switch (existingDataCheck.type) {
             case "no-values":
-                return save();
+                return save(loading);
```

Comment:
There is another save in `saveAndSendExistingDataMessage`, use loading also there (thus, the `loading` argument of `save`  won't be optional)